pkg/pki/ssh/verify.go

```diff
@@ -0,0 +1,35 @@
+package ssh
+
+import (
+	"crypto/sha512"
+
+	"golang.org/x/crypto/ssh"
+)
+
+func Verify(message []byte, armoredSignature string, publicKey []byte) error {
+	decodedSignature, _, err := Decode(armoredSignature)
+	if err != nil {
+		return err
+	}
+
+	desiredPk, _, _, _, err := ssh.ParseAuthorizedKey(publicKey)
+	if err != nil {
+		return err
+	}
+
+	// Hash the message so we can verify it against the signature.
+	h := sha512.New()
```

Comment:
`io.Copy` from reader straight to writer `&h`

Reply:
Done!